### DISCUSSION TITLE
Adding nullValueIterableMappingStrategy and nullValueMapMappingStrategy to MapStruct processor options

### DISCUSSION THREAD

**Discussion by shanon84** [mapstruct/mapstruct#3319] [Ideas]
Hi there,
currently we cannot add nullValueIterableMappingStrategy and nullValueMapMappingStrategy as processor options (only in mapper and mapper config). It would be nice to be able to change the default of each strategies globally.

**Comment by filiphr** [comment]
I agree @shanon84. Adding those strategies as options would be good. Are you interested in working on that?

**Comment by shanon84** [comment]
@filiphr mh, I checked out the 1.6.0-SNAPSHOT. It is already implemented and works. Thats why I was confused and could not find the problem while searching for it.
But I maybe found a bug:
Lets say you have an Adress and a Person model you want to map. Adress has a List of "persons". The generated mappers will set an empty list for persons, if the list is null -> as aspected.
The problem is, if you have a List<String> member in Adress: this one will still be set to null, if the source value is null.
Or is this somehow as it should be?

**Reply by filiphr** [reply]
Thanks for checking it out, I was not 100% sure if we have it or not, and I didn't have my IDE open.

For your other question, best would be to create some examples that we can try and see what is happening, on first thought I think that it should work like you are saying, but it might be something else

**Reply by shanon84** [reply]
I forked mapstruct and commited 2 model changes and a test.
https://github.com/shanon84/mapstruct/commit/89629fac02d91135cc6f4961f6c1e6c361cf497e

**Reply by filiphr** [reply]
Thanks @shanon84, I think that your issue is related to https://github.com/mapstruct/mapstruct/issues/2843. We currently apply the strategy only to generated methods and not to mappings, it does lead to some inconsistencies I agree
